### PR TITLE
Co2 emission

### PR DESCRIPTION
**Pull request by azime**
Manage co2 emission

### CODE REVIEW

File: source/ed/connectors/fusio_parser.h [hunk 1]
```diff
@@ -180,7 +180,7 @@ struct CompanyFusioHandler : public GenericHandler {
 
 struct PhysicalModeFusioHandler : public GenericHandler {
     PhysicalModeFusioHandler(GtfsData& gdata, CsvReader& reader) : GenericHandler(gdata, reader) {}
-    int id_c, name_c;
+    int id_c, name_c, co2_emission;
```
Comment (l-vincent-l) [inline source/ed/connectors/fusio_parser.h]: maybe you can rename in co2_emission_c

File: source/routing/raptor_api.cpp [hunk 2]
```diff
@@ -110,6 +110,9 @@ static void fill_section(pbnavitia::Section *pb_section, const type::VehicleJour
         add_coord(stop_times.front()->journey_pattern_point->stop_point->coord, pb_section);
         add_coord(stop_times.back()->journey_pattern_point->stop_point->coord, pb_section);
     }
+    if (vj && vj->journey_pattern && vj->journey_pattern->physical_mode){
+        fill_co2_emission(pb_section, vj->journey_pattern->physical_mode);
```
Comment (l-vincent-l) [inline source/routing/raptor_api.cpp]: move it into fill_section, so we ensure it's always fill

File: source/routing/raptor_api.cpp [hunk 3]
```diff
@@ -238,7 +241,7 @@ static void add_pathes(EnhancedResponse& enhanced_response,
                     if(i > 0) {
                         const auto & previous_coord = item.stop_points[i-1]->coord;
                         const auto & current_coord = item.stop_points[i]->coord;
-                        length += previous_coord.distance_to(current_coord);
+                        length += previous_coord.distance_to(current_coord);
```
Comment (l-vincent-l) [inline source/routing/raptor_api.cpp]: ARGGGGGGGGGG trailing spaces!

**Comment by azime** [comment]
corrections

File: source/routing/raptor_api.cpp (hunk 3 above)
Comment (TeXitoi) [inline source/routing/raptor_api.cpp]: unwanted spaces

File: source/routing/raptor_api.cpp [hunk 4]
```diff
@@ -260,7 +263,7 @@ static void add_pathes(EnhancedResponse& enhanced_response,
                 }
                 pb_section->set_length(length);
                 bt::time_period action_period(departure_ptime, arrival_ptime);
-                fill_section(pb_section, vj, item.stop_times, d, now, action_period);
+                fill_section(pb_section, vj, item.stop_times, d, now, action_period);
```
Comment (TeXitoi) [inline source/routing/raptor_api.cpp]: unwanted spaces

File: source/ed/connectors/fusio_parser.h (hunk 1 above)
Comment (TeXitoi) [inline source/ed/connectors/fusio_parser.h]: by conventions, column idx ends with _c

File: source/ed/types.h [hunk 5]
```diff
@@ -146,6 +146,7 @@ struct CommercialMode : public Header, Nameable{
 
 struct PhysicalMode : public Header, Nameable{
     const static nt::Type_e type = nt::Type_e::PhysicalMode;
+    double co2_emission;
```
Comment (TeXitoi) [inline source/ed/types.h]: to be sure this field is correctly initialized by the default constructor, you can write this line

``` c++
    double co2_emission = 0.;
```

File: source/type/pb_converter.cpp [hunk 6]
```diff
@@ -863,7 +863,6 @@ static void finalize_section(pbnavitia::Section* section,
 
     section->set_begin_date_time(navitia::to_posix_timestamp(departure));
     section->set_end_date_time(navitia::to_posix_timestamp(departure + bt::seconds(total_duration)));
-
```
Comment (TeXitoi) [inline source/type/pb_converter.cpp:866]: I prefere with the empty line

File: source/type/pb_converter.cpp [hunk 7]
```diff
@@ -1010,7 +1009,6 @@ void fill_crowfly_section(const type::EntryPoint& origin, const type::EntryPoint
     section->set_length(0);
     section->set_end_date_time(navitia::to_posix_timestamp(time));
     section->set_type(pbnavitia::SectionType::CROW_FLY);
-
```
Comment (TeXitoi) [inline source/type/pb_converter.cpp:1013]: I prefere with the empty line

File: source/type/type.h [hunk 8]
```diff
@@ -489,10 +489,11 @@ struct CommercialMode : public Header, Nameable{
 
 struct PhysicalMode : public Header, Nameable{
     const static Type_e type = Type_e::PhysicalMode;
+    double co2_emission;
```
Comment (TeXitoi) [inline source/type/type.h]: same remark as for ed::type, add `= 0.` to the declaration

**Comment by azime** [comment]
corrections

File: source/type/type.h [hunk 9]
```diff
@@ -489,10 +489,11 @@ struct CommercialMode : public Header, Nameable{
...
     std::vector<JourneyPattern*> journey_pattern_list;
 
     template<class Archive> void serialize(Archive & ar, const unsigned int ) {
-        ar & idx & name & uri & journey_pattern_list;
+        ar & idx & name & uri & co2_emission & journey_pattern_list;
```
Comment (kinnou02) [inline source/type/type.h:496]: you need to upgrade the version of the nav

**Comment by azime** [comment]
data version

**Comment by TeXitoi** [comment]
You still have 2 comments by me that is not fixed :'(

**Comment by azime** [comment]
empty line added

**Comment by azime** [comment]
initialize variable in constructor

**Comment by TeXitoi** [comment]
[15:23:46] ‎gpinot‎: bon, je pense que je n'ai pas été clair dans mes commentaires ;)
‎[15:24:03] ‎gpinot‎: c'était bien comme c'était dans ed/types.h
‎[15:24:33] ‎gpinot‎: tu peux même enlever la ligne du constructeur PhysicalMode() {}
‎[15:24:52] ‎gpinot‎: je demandais juste de faire pareil dans type/type.h
‎[15:25:11] ‎gpinot‎: de mettre double co2_emission = 0.;
‎[15:25:19] ‎gpinot‎: désolé si j'ai pas été clair

**Comment by azime** [comment]
initialize co2_emission variable

**Comment by thiphariel** [comment]
Problem : co2_emission is never attached on a "vehicle_journey" object

**Pull request merged**